### PR TITLE
Stylus V2 New IA

### PR DESCRIPTION
**Pull request by srinjoyc**
# Stylus Documentation Update

This update reorganizes the Stylus section of the documentation to improve structure and follow conventional information architecture (IA). The changes minimally modify existing content while integrating new content from SBE.

## **Updates**
- **Introduction**: See [PR #1782](https://github.com/OffchainLabs/arbitrum-docs/pull/1782) for specific updates.
- SBE script to pull in application examples as well as the basic examples from before.

## **Removes**
- `cli-tools-overview`
- "Public Preview" mentions from Stylus docs and the `public-preview` doc. Some references remain and should eventually be removed.

### Moved Content:
- **Arbiscan Contract Verification**: Moved to **Tutorial**.
- **Other Supported Languages**: Moved to **Tutorial**.
- **Run a Stylus Dev Node**: Moved to **Tutorial**.
- **Troubleshooting**: Moved to **Reference**.
- **Source Code Repository**: Moved to **Reference**.
- **Concepts from Gentle Introduction**: Now under **How Stylus Works**.
- **Gas and Ink Overview**: Now under **Concepts** > **How Stylus Uses Gas**.

## **New Additions**
- **Awesome Stylus Link**: Added to **Examples**.
- **Call** basic example from SBE added to Rust SDK.
- New examples added:
  - **ERC20**
  - **ERC721**
  - **Multi-Call**
  - **Vending Machine**

### CODE REVIEW

**Comment by srinjoyc** [comment]
I ran `yarn format` in the website directory  and tried manually running prettier on the file - it seems to product no warnings / errors locally. Not sure what the issue is..

**Review by joshuacolvin0 — APPROVED** [review]

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+title: 'How Stylus works'
+sidebar_label: 'How Stylus works'
+description: 'Learn what powers Stylus'
+author: srinjoyc
+SME: srinjoyc
+user_story: As an Ethereum developer/project owner, I need to vet the Stylus.
+content_type: concept
+---
+
+There are four main steps for bringing a Stylus program to life: <b>coding, activation, execution, and proving.</b>
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/concepts/how-it-works.md:12]:
```suggestion
There are four main steps for bringing a Stylus program to life: **coding, activation, execution, and proving**.
```

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+### Coding
+
+Developers can now write smart contracts in any programming language that compiles to WASM. Note that some high-level languages generate far more performant WASMs than others.
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/concepts/how-it-works.md:16]:
```suggestion
Developers can now write smart contracts in any programming language that compiles to WASM.

:::note
Some high-level languages generate far more performant WASMs than others.
:::
```

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+Initially, there will be support for Rust, C, and C++. However, the levels of support will differ at first. Rust has rich language support from day one, with an open-source SDK that makes writing smart contracts in Rust as easy as possible. C and C++ are supported off the bat, too, enabling the deployment of existing contracts in those languages on-chain with minimal modifications.
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/concepts/how-it-works.md:18]:
```suggestion
Initially, there will be support for Rust, C, and C++. However, the levels of support will vary. Rust has rich language support from day one, with an open-source SDK that makes writing smart contracts in Rust as easy as possible. C and C++ are supported off the bat, enabling the deployment of existing contracts in those languages on-chain with minimal modifications.
```

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+The Stylus SDK for Rust contains the smart contract development framework and language features most developers will need to use in Stylus. The SDK also makes it possible to perform all of the EVM-specific functionalities that smart contract developers are used to. Check out the [Rust SDK Guide](https://docs.arbitrum.io/stylus/rust-sdk-guide) and the [Crate Docs](https://docs.rs/stylus-sdk/latest/stylus_sdk/index.html).
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/concepts/how-it-works.md:20]:
```suggestion
The Stylus SDK for Rust contains the smart contract development framework and language features most developers will need to use in Stylus. The SDK also makes it possible to perform all EVM-specific functionalities that smart contract developers use. Check out the [Rust SDK Guide](https://docs.arbitrum.io/stylus/rust-sdk-guide) and the [Crate Docs](https://docs.rs/stylus-sdk/latest/stylus_sdk/index.html).
```

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+### Activation
+
+Stylus programs are compiled to WASM and then lowered to assembly. Starting from a high-level language (such as Rust, C, or C++), the first compilation stage happens either using the CLI provided in the Stylus SDK for Rust or any other compiler, such as Clang for C and C++. Once compiled, the WASM is posted onchain. Then, in a process called activation, WASM gets lowered to a node's native machine code (such as ARM or x86).
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/concepts/how-it-works.md:24]:
```suggestion
Stylus programs are compiled to WASM, and then lowered to assembly. Starting from a high-level language (such as Rust, C, or C++), the first compilation stage happens using the CLI provided in the Stylus SDK for Rust or any other compiler, such as Clang for C and C++. Once compiled, the WASM is posted onchain. Then, in an activation process, WASM gets lowered to a node's native machine code (such as ARM or x86).
```

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+Activating a Stylus program requires a new precompile, ArbWasm. This precompile produces efficient binary code tailored to a node's native assembly. During this step, a series of middlewares ensure user programs can be safely executed and deterministically fraud-proven. Instrumentation includes gas metering, depth-checking, memory charging, and more to guarantee all WASM programs are safe for the chain to execute. Stylus contracts can be called only after they've been activated.
+
+Gas metering is essential for certifying that computational resources are paid for. In Stylus, the unit for measuring cost is called “ink,” similar to Ethereum's gas but thousands of times smaller. There are two reasons why a new measurement is used: First, WASM execution is so much faster than the EVM that thousands of WASM opcodes could be executed in the same time it takes the EVM to execute one. Second, the conversion rate of ink to gas can change based on future hardware or VM improvements. For a conceptual introduction to Stylus gas and ink, see [gas and ink (Stylus)](https://docs.arbitrum.io/stylus/concepts/stylus-gas).
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/concepts/how-it-works.md:28]:
```suggestion
Gas metering is essential for certifying that computational resources are paid for. In Stylus, the unit for measuring cost is called **ink**, which is similar to Ethereum's gas but thousands of times smaller. There are two reasons why a new measurement is used: First, WASM execution is so much faster than the EVM that executing thousands of WASM opcodes could be done in the same amount of time it takes the EVM to execute one. Second, the conversion rate of ink to gas can change based on future hardware or VM improvements. For a conceptual introduction to Stylus gas and ink, see [gas and ink (Stylus)](https://docs.arbitrum.io/stylus/concepts/stylus-gas).
```

Comment (pete-vielhaber) [reply arbitrum-docs/stylus/concepts/how-it-works.md:28]: I would suggest defining ink and linking to glossary

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+### Execution
+
+Stylus programs are executed in a fork of [Wasmer](https://wasmer.io/), the leading WebAssembly runtime, with minimal changes to optimize their codebase for blockchain-specific use cases. Wasmer executes native code much faster than <a data-quicklook-from="geth">Geth</a> executes EVM bytecode, contributing to the significant gas savings that Stylus provides.
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/concepts/how-it-works.md:32]:
```suggestion
Stylus programs execute in a fork of [Wasmer](https://wasmer.io/), the leading WebAssembly runtime, with minimal changes to optimize their codebase for blockchain-specific use cases. Wasmer executes native code much faster than <a data-quicklook-from="geth">Geth</a> executes EVM bytecode, contributing to the significant gas savings that Stylus provides.
```

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+EVM contracts continue to execute the same way they did before Stylus. When a contract is called, the difference between an EVM contract and a WASM program can be seen via an [EOF](https://notes.ethereum.org/@ipsilon/evm-object-format-overview)-inspired contract header. From there, the contract is executed using its corresponding runtime. Contracts written in Solidity and WASM languages can make cross-contract calls to each other, meaning a developer never has to consider what language the contract was written in. Everything is interoperable.
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/concepts/how-it-works.md:34]:
```suggestion
EVM contracts continue to execute the same way they were before Stylus. When calling a contract, the difference between an EVM contract and a WASM program is visible via an [EOF](https://notes.ethereum.org/@ipsilon/evm-object-format-overview)-inspired contract header. From there, the contract executes using its corresponding runtime. Contracts written in Solidity and WASM languages can make cross-contract calls to each other, meaning a developer never has to consider which language the contract is in. Everything is interoperable.
```

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+For more details on Nitro’s architecture, refer to the [documentation](/how-arbitrum-works/inside-arbitrum-nitro.mdx) or the [Nitro whitepaper](https://github.com/OffchainLabs/nitro/blob/master/docs/Nitro-whitepaper.pdf).
+
+### Why does this matter?
+
+Stylus innovates on many levels, with the key ones described here:
+
+#### One chain, many languages
+
+There are estimated to be roughly 20k Solidity developers, compared to 3 million Rust developers or 12 million C developers [[1](https://slashdatahq.medium.com/state-of-the-developer-nation-23rd-edition-the-fall-of-web-frameworks-coding-languages-711525e3df3a)]. Developers are now free to use their preferred programming language, all interoperable on any <a data-quicklook-from="arbitrum-chain">Arbitrum chain</a> with Stylus. By onboarding the next million developers, scaling to the next billion users becomes possible.
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/concepts/how-it-works.md:50]:
```suggestion
There are roughly 20k Solidity developers, compared to 3 million Rust developers or 12 million C developers [[1](https://slashdatahq.medium.com/state-of-the-developer-nation-23rd-edition-the-fall-of-web-frameworks-coding-languages-711525e3df3a)]. Developers can now use their preferred programming language, which is  interoperable on any <a data-quicklook-from="arbitrum-chain">Arbitrum chain</a> with Stylus. By onboarding the next million developers, scaling to the next billion users becomes possible.
```

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+Stylus' MultiVM brings the best of both worlds. Developers still get all of the benefits of the EVM, including the ecosystem and liquidity, while getting efficiency improvements and access to existing libraries in Rust, C, and C++, all without changing anything about how the EVM works. EVM equivalence is no longer the ceiling; it's the floor.
+
+#### Cheaper execution
+
+Stylus is a more efficient execution environment than the EVM, leading directly to gas savings for complex smart contracts. Computation and memory can be significantly cheaper. Cryptography libraries can be permissionlessly deployed as custom application layer precompiles. Use cases that are impractical in the EVM are now possible in Stylus.
+
+#### Opt-in reentrancy
+
+Stylus doesn't just improve on cost and speed. WASM programs are also safer. Reentrancy is a common vulnerability that developers can only attempt to mitigate in Solidity. Stylus provides cheap reentrancy detection, and using the Rust SDK, reentrancy is disabled by default unless intentionally overridden.
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/concepts/how-it-works.md:62]:
```suggestion
Stylus doesn't just improve on cost and speed. WASM programs are also safer. Reentrancy is a common vulnerability developers can only attempt to mitigate in Solidity. Stylus provides cheap reentrancy detection, and using the Rust SDK, reentrancy is disabled by default unless intentionally overridden.
```

File: arbitrum-docs/stylus/stylus-gentle-introduction.md [hunk 11]
```diff
@@ -9,119 +9,45 @@ sidebar_position: 1
...
-- Bringing existing games written in C++ on-chain
-- Compute-heavy AI models
-- Projects using alternative signature schemes, such as `secp256r1`, via custom precompiles
-- Optimization of Solidity-based projects for speed and cost
-
-We leave it up to you to ideate blockchain projects that were not technically feasible until now.
+With a WASM VM, any programming language that can compile to WASM is within Stylus's scope. While many popular programming languages can be compiled into WASM, some compilers are more suitable for smart contract development than others, like Rust, C, and C++. Other languages like Go, Sway, Move, and Cairo can also be supported. Languages that include their own runtimes, like Python and Javascript, are more complex for Stylus to support, although not impossible. Compared to Solidity, WASM programs are much more efficient for memory-intensive applications. There are many reasons for this, including the decades of compiler development for Rust and C. WASM also has a faster runtime than the EVM, resulting in faster execution. Third-party [contribution](#contributing) in the form of libraries for new and existing languages is welcomed!
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/stylus-gentle-introduction.md:26]:
```suggestion
With a WASM VM, any programming language compilable to WASM is within Stylus's scope. While many popular programming languages can compile into WASM, some compilers are more suitable for smart contract development than others, like Rust, C, and C++. Other languages like Go, Sway, Move, and Cairo are also  supported. Languages that include their own runtimes, like Python and Javascript, are more complex for Stylus to support, although not impossible. Compared to Solidity, WASM programs are much more efficient for memory-intensive applications. There are many reasons for this, including the decades of compiler development for Rust and C. WASM also has a faster runtime than the EVM, resulting in faster execution. Third-party [contribution](#contributing) in the form of libraries for new and existing languages is welcomed!
```

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+description: 'Learn what powers Stylus'
+author: srinjoyc
+SME: srinjoyc
+user_story: As an Ethereum developer/project owner, I need to vet the Stylus.
+content_type: concept
+---
+
+There are four main steps for bringing a Stylus program to life: <b>coding, activation, execution, and proving.</b>
+
+### Coding
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/concepts/how-it-works.md:14]:
```suggestion
## Coding
```

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+
+Developers can now write smart contracts in any programming language that compiles to WASM. Note that some high-level languages generate far more performant WASMs than others.
+
+Initially, there will be support for Rust, C, and C++. However, the levels of support will differ at first. Rust has rich language support from day one, with an open-source SDK that makes writing smart contracts in Rust as easy as possible. C and C++ are supported off the bat, too, enabling the deployment of existing contracts in those languages on-chain with minimal modifications.
+
+The Stylus SDK for Rust contains the smart contract development framework and language features most developers will need to use in Stylus. The SDK also makes it possible to perform all of the EVM-specific functionalities that smart contract developers are used to. Check out the [Rust SDK Guide](https://docs.arbitrum.io/stylus/rust-sdk-guide) and the [Crate Docs](https://docs.rs/stylus-sdk/latest/stylus_sdk/index.html).
+
+### Activation
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/concepts/how-it-works.md:22]:
```suggestion
## Activation
```

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+
+Activating a Stylus program requires a new precompile, ArbWasm. This precompile produces efficient binary code tailored to a node's native assembly. During this step, a series of middlewares ensure user programs can be safely executed and deterministically fraud-proven. Instrumentation includes gas metering, depth-checking, memory charging, and more to guarantee all WASM programs are safe for the chain to execute. Stylus contracts can be called only after they've been activated.
+
+Gas metering is essential for certifying that computational resources are paid for. In Stylus, the unit for measuring cost is called “ink,” similar to Ethereum's gas but thousands of times smaller. There are two reasons why a new measurement is used: First, WASM execution is so much faster than the EVM that thousands of WASM opcodes could be executed in the same time it takes the EVM to execute one. Second, the conversion rate of ink to gas can change based on future hardware or VM improvements. For a conceptual introduction to Stylus gas and ink, see [gas and ink (Stylus)](https://docs.arbitrum.io/stylus/concepts/stylus-gas).
+
+### Execution
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/concepts/how-it-works.md:30]:
```suggestion
## Execution
```

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+
+### Proving
+
+Nitro operates in two modes: a "happy case" where it compiles execution history to native code, and a "sad case" during validator disputes, where it compiles execution history to WASM for interactive fraud proofs on Ethereum. Stylus builds on Nitro's fraud-proving technology, allowing it to verify both execution history and WASM programs deployed by developers.
+
+Stylus is made possible by Nitro’s ability to replay and verify disputes using WASM. Validators bisect disputes until an invalid step is identified and proven on-chain through a [“one-step proof.”](/how-arbitrum-works/fraud-proofs/challenge-manager.mdx#general-bisection-protocol). This deterministic fraud-proving capability ensures the correctness of any arbitrary program compiled to WASM. The combination of WASM's and Nitro's properties enables this technological leap we call Stylus.
+
+For more details on Nitro’s architecture, refer to the [documentation](/how-arbitrum-works/inside-arbitrum-nitro.mdx) or the [Nitro whitepaper](https://github.com/OffchainLabs/nitro/blob/master/docs/Nitro-whitepaper.pdf).
+
+### Why does this matter?
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/concepts/how-it-works.md:44]:
```suggestion
## Why does this matter?
```

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+
+Stylus innovates on many levels, with the key ones described here:
+
+#### One chain, many languages
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/concepts/how-it-works.md:48]:
```suggestion
### One chain, many languages
```

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+
+There are estimated to be roughly 20k Solidity developers, compared to 3 million Rust developers or 12 million C developers [[1](https://slashdatahq.medium.com/state-of-the-developer-nation-23rd-edition-the-fall-of-web-frameworks-coding-languages-711525e3df3a)]. Developers are now free to use their preferred programming language, all interoperable on any <a data-quicklook-from="arbitrum-chain">Arbitrum chain</a> with Stylus. By onboarding the next million developers, scaling to the next billion users becomes possible.
+
+#### A better EVM
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/concepts/how-it-works.md:52]:
```suggestion
### A better EVM
```

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 18]
```diff
@@ -0,0 +1,66 @@
...
+
+Stylus' MultiVM brings the best of both worlds. Developers still get all of the benefits of the EVM, including the ecosystem and liquidity, while getting efficiency improvements and access to existing libraries in Rust, C, and C++, all without changing anything about how the EVM works. EVM equivalence is no longer the ceiling; it's the floor.
+
+#### Cheaper execution
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/concepts/how-it-works.md:56]:
```suggestion
### Cheaper execution
```

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 19]
```diff
@@ -0,0 +1,66 @@
...
+
+Stylus is a more efficient execution environment than the EVM, leading directly to gas savings for complex smart contracts. Computation and memory can be significantly cheaper. Cryptography libraries can be permissionlessly deployed as custom application layer precompiles. Use cases that are impractical in the EVM are now possible in Stylus.
+
+#### Opt-in reentrancy
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/concepts/how-it-works.md:60]:
```suggestion
### Opt-in reentrancy
```

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 20]
```diff
@@ -0,0 +1,66 @@
...
+
+Stylus doesn't just improve on cost and speed. WASM programs are also safer. Reentrancy is a common vulnerability that developers can only attempt to mitigate in Solidity. Stylus provides cheap reentrancy detection, and using the Rust SDK, reentrancy is disabled by default unless intentionally overridden.
+
+#### Fully interoperable
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/concepts/how-it-works.md:64]:
```suggestion
### Fully interoperable
```

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 21]
```diff
@@ -0,0 +1,66 @@
...
+
+#### A better EVM
+
+Stylus' MultiVM brings the best of both worlds. Developers still get all of the benefits of the EVM, including the ecosystem and liquidity, while getting efficiency improvements and access to existing libraries in Rust, C, and C++, all without changing anything about how the EVM works. EVM equivalence is no longer the ceiling; it's the floor.
+
+#### Cheaper execution
+
+Stylus is a more efficient execution environment than the EVM, leading directly to gas savings for complex smart contracts. Computation and memory can be significantly cheaper. Cryptography libraries can be permissionlessly deployed as custom application layer precompiles. Use cases that are impractical in the EVM are now possible in Stylus.
```
Comment (pete-vielhaber) [inline arbitrum-docs/stylus/concepts/how-it-works.md:58]:
```suggestion
Stylus is a more efficient execution environment than the EVM, leading directly to gas savings for complex smart contracts. Computation and memory can be significantly cheaper. Deploying cryptography libraries can be done permissionlessly as custom application layer precompiles. Use cases that are impractical in the EVM are now possible in Stylus.
```

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 22]
```diff
@@ -9,58 +9,62 @@ user_story: As an Ethereum developer/project owner, I need to vet the Stylus.
...
+:::note
+Some high-level languages generate far more performant WASMs than others.
+:::
 
-The Stylus SDK for Rust contains the smart contract development framework and language features most developers will need to use in Stylus. The SDK also makes it possible to perform all of the EVM-specific functionalities that smart contract developers are used to. Check out the [Rust SDK Guide](https://docs.arbitrum.io/stylus/rust-sdk-guide) and the [Crate Docs](https://docs.rs/stylus-sdk/latest/stylus_sdk/index.html).
+Initially, there will be support for Rust, C, and C++. However, the levels of support will vary. Rust has rich language support from day one, with an open-source SDK that makes writing smart contracts in Rust as easy as possible. C and C++ are supported off the bat, enabling the deployment of existing contracts in those languages on-chain with minimal modifications.
```
Comment (joshuacolvin0) [inline arbitrum-docs/stylus/concepts/how-it-works.md:22]:
```suggestion
Currently there is support for Rust, C, and C++. However, the levels of support varies. Rust has rich language support from day one, with an open-source SDK that makes writing smart contracts in Rust as easy as possible. C and C++ is supported off the bat, enabling the deployment of existing contracts in those languages on-chain with minimal modifications.
```

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 23]
```diff
@@ -9,58 +9,62 @@ user_story: As an Ethereum developer/project owner, I need to vet the Stylus.
...
+## Activation
 
-Activating a Stylus program requires a new precompile, ArbWasm. This precompile produces efficient binary code tailored to a node's native assembly. During this step, a series of middlewares ensure user programs can be safely executed and deterministically fraud-proven. Instrumentation includes gas metering, depth-checking, memory charging, and more to guarantee all WASM programs are safe for the chain to execute. Stylus contracts can be called only after they've been activated.
+Stylus programs are compiled to WASM, and then lowered to assembly. Starting from a high-level language (such as Rust, C, or C++), the first compilation stage happens using the CLI provided in the Stylus SDK for Rust or any other compiler, such as Clang for C and C++. Once compiled, the WASM is posted onchain. Then, in an activation process, WASM gets lowered to a node's native machine code (such as ARM or x86).
```
Comment (joshuacolvin0) [inline arbitrum-docs/stylus/concepts/how-it-works.md:28]:
```suggestion
Starting from a high-level language (such as Rust, C, or C++), the first compilation stage happens using the CLI provided in the Stylus SDK for Rust or any other compiler, such as Clang for C and C++. Once compiled, the WASM is posted onchain. Then, in an activation process, WASM gets lowered to a node's native machine code (such as ARM or x86).
```

**Review by joshuacolvin0 — APPROVED** [review]

**Review by joshuacolvin0 — APPROVED** [review]
LGTM

**Review by pete-vielhaber — APPROVED** [review]
- Made a single suggestion

File: arbitrum-docs/stylus/concepts/how-it-works.md [hunk 24]
```diff
@@ -0,0 +1,70 @@
...
+
+Developers can now write smart contracts in any programming language that compiles to WASM.
+
+:::note
+Some high-level languages generate far more performant WASMs than others.
+:::
+
+Currently there is support for Rust, C, and C++. However, the levels of support varies. Rust has rich language support from day one, with an open-source SDK that makes writing smart contracts in Rust as easy as possible. C and C++ is supported off the bat, enabling the deployment of existing contracts in those languages on-chain with minimal modifications.
+
+The Stylus SDK for Rust contains the smart contract development framework and language features most developers will need to use in Stylus. The SDK also makes it possible to perform all EVM-specific functionalities that smart contract developers use. Check out the [Rust SDK Guide](https://docs.arbitrum.io/stylus/rust-sdk-guide) and the [Crate Docs](https://docs.rs/stylus-sdk/latest/stylus_sdk/index.html).
```
Comment (anegg0) [inline arbitrum-docs/stylus/concepts/how-it-works.md:24]:
```suggestion
Developers can now write smart contracts in any programming language that compiles to WASM.

:::note
Some high-level languages generate far more performant WASMs than others.
:::

Currently, there is support for Rust, C, and C++. However, the levels of support vary. Rust has rich language support from day one, with an open-source SDK that makes writing smart contracts in Rust as easy as possible. C and C++ are supported off the bat, enabling the deployment of existing contracts in those languages on-chain with minimal modifications.

The Stylus SDK for Rust contains the smart contract development framework and language features most developers will need to use in Stylus. The SDK also makes it possible to perform all EVM-specific functionalities that smart contract developers use. Check out the [Rust SDK Guide](https://docs.arbitrum.io/stylus/rust-sdk-guide) and the [Crate Docs](https://docs.rs/stylus-sdk/latest/stylus_sdk/index.html).
```

**Review by anegg0 — CHANGES_REQUESTED** [review]
I suggested a few grammar fixes. Once those have been addressed, it all LGTM.

**Review by anegg0 — APPROVED** [review]

**Pull request merged**